components/admin_console/plugin_management/plugin_management.jsx

```diff
@@ -459,10 +461,37 @@ export default class PluginManagement extends AdminSettings {
 
         this.setState({uploading: true});
 
-        const {error} = await this.props.actions.uploadPlugin(file);
-        this.setState({fileSelected: false, fileName: null, uploading: false, serverError: null});
+        const {error} = await this.props.actions.uploadPlugin(file, false);
+        this.setState({uploading: false, serverError: null});
         Utils.clearFileInput(element);
 
+        if (error) {
+            if (error.server_error_id === 'app.plugin.install_id.app_error') {
+                this.setState({showOverwritePluginModal: true});
+                return;
+            }
+            this.setState({file: null, fileSelected: false});
+
+            if (error.server_error_id === 'app.plugin.activate.app_error') {
```

Comment:
This code duplication is unfortunate. Could we instead splice `handleSubmitUpload` into a second method, `handleUpload`, that accepts the file in question and a `force` parameter of its own?

Reply:
done.